Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -4156,17 +4291,26 @@ - (void)resetMappings
         // We need to `beginLongLivedReadTransaction` before we update our
         // mapping in order to jump to the most recent commit.
         [self.uiDatabaseConnection beginLongLivedReadTransaction];
+        [self extendRangeToIncludeUnobservedItems];
         [self.uiDatabaseConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
             [self.messageMappings updateWithTransaction:transaction];
         }];
-        [self updateMessageMappingRangeOptions:MessagesRangeSizeMode_Normal];
+        [self updateMessageMappingRangeOptions];
     }
     [self reloadViewItems];
 
     [self resetContentAndLayout];
     [self ensureDynamicInteractions];
     [self updateBackButtonUnreadCount];
     [self updateNavigationBarSubtitleLabel];
+
+    // There appears to be a bug in YapDatabase that sometimes delays modifications
+    // made in another process (e.g. the SAE) from showing up in other processes.
+    // There's a simple workaround: a trivial write to the database flushes changes
+    // made from other processes.
+    [self.editingDatabaseConnection setObject:[NSUUID UUID].UUIDString
+                                       forKey:@"conversation_view_noop_mod"
+                                 inCollection:@"temp"];
```

Comment:
This turned out to be key change.

Reply:
It would be good to make this async to avoid blocking the UI when other writes are happening.

But would it still solve our problem? I *think* it would because ultimately we're relying on the a new dbModified notification to fix up the state of the uiDatabase.

Can you also make a card to follow up with upstream about this bug?

TIOLI

Reply:
Done.